chalk-ir/src/lib.rs

```diff
@@ -1413,21 +1512,21 @@ where
 /// distinctions.
 ///
 /// To produce one of these values, use the `u_canonicalize` method.
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct UCanonical<T> {
-    pub canonical: Canonical<T>,
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct UCanonical<I: Interner, T> {
```

Comment:
As above, we should add `T: HasInterner`

Reply:
Done.